src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -322,6 +332,76 @@ private async Task IndexAddressesContinuouslyAsync()
+            var previousTransactionsToRetrieve = new HashSet<uint256>();
+
+            foreach (Transaction tx in blockToReorg.Transactions)
+            {
+                if (tx.IsCoinBase)
+                    continue;
+
+                foreach (TxIn input in tx.Inputs)
+                {
+                    previousTransactionsToRetrieve.Add(input.PrevOut.Hash);
+                }
+            }
+
+            // This is highly unlikely in practice outside of a unit test.
+            if (previousTransactionsToRetrieve.Count == 0)
+            {
+                return;
+            }
+
+            Transaction[] previousTransactionArray = this.blockStore.GetTransactionsByIds(previousTransactionsToRetrieve.ToArray());
+
+            if (previousTransactionArray == null)
+            {
+                this.logger.LogError("Missing previous transaction data for block {0}.", blockToReorg.GetHash());
+                this.logger.LogTrace("(-)[MISSING_TRANSACTION_DATA]");
+                throw new Exception($"Unable to retrieve previous transaction data for block {blockToReorg.GetHash()}.");
+            }
+
+            var previousTransactions = new Dictionary<uint256, Transaction>();
+
+            foreach (Transaction tx in previousTransactionArray)
+            {
+                previousTransactions.TryAdd(tx.GetHash(), tx);
+            }
+
+            // Now unwind the effect of each transaction on the outpoint repository.
+            foreach (Transaction tx in blockToReorg.Transactions)
+            {
+                foreach (TxIn input in tx.Inputs)
+                {
+                    OutPoint previouslyConsumedOutput = input.PrevOut;
+
+                    // Look up previous transaction to determine amount.
+                    previousTransactions.TryGetValue(previouslyConsumedOutput.Hash, out Transaction previousTransaction);
+
+                    if (previousTransaction == null)
+                    {
+                        this.logger.LogError("Missing transaction data for {0}.", previouslyConsumedOutput.Hash);
+                        this.logger.LogTrace("(-)[PREVIOUS_TRANSACTION_MISSING]");
+                        throw new Exception($"Unable to retrieve transaction data for {previouslyConsumedOutput.Hash}.");
+                    }
+
+                    var reconstructedOutPointData = new OutPointData()
+                    {
+                        Outpoint = previouslyConsumedOutput.ToString(),
+                        ScriptPubKeyBytes = previousTransaction.Outputs[previouslyConsumedOutput.N].ScriptPubKey.ToBytes(),
+                        Money = previousTransaction.Outputs[previouslyConsumedOutput.N].Value
+                    };
+
+                    // Place outpoint that is now unspent back in the repository.
+                    this.outpointsRepository.AddOutPointData(reconstructedOutPointData);
```

Comment:
So does this imply that at this point we have already removed the reorg outpoint?

Reply:
Yes, it would have been removed when the block was initially processed by the indexer.